Docs: Added section about the `watchdog` and  `error` properties. (#6335)
ckeditor/ckeditor5: docs/builds/guides/frameworks/angular.md

</button>
```

### `watchdog`

> An instance of the {@link module:watchdog/contextwatchdog~ContextWatchdog `ContextWatchdog`} class that is responsible for providing the same context to the editor instances and restarting the whole structure in case of crashes.

Reviewer: ```suggestion
An instance of the {@link module:watchdog/contextwatchdog~ContextWatchdog `ContextWatchdog`} class that is responsible for providing the same context to multiple editor instances and restarting the whole structure in case of crashes.
```

Fired when the editor is ready. It corresponds with the [`editor#ready`](https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editor-Editor.html#event-ready) event.
It is fired with the editor instance.

> Note that this method might be called multiple times. It is called whenever the editor crashes and is restarted, hence the editor property should not be kept internally (because it might change).

Reviewer: ```suggestion
Note that this method might be called multiple times. Apart from initialization, it is also called whenever the editor is restarted after a crash. Do not keep the reference to the editor instance internally, because it will change in case of restart. Instead, you should use `watchdog.editor` property.
```

Fired when the editing view of the editor is focused. It corresponds with the {@link module:engine/view/document~Document#event:focus `editor.editing.view.document#focus`} event.
It is fired with an object containing the editor and the CKEditor 5 `focus` event data.

### `error`

> Fired when the editor crashes after its initialization. Once the editor is crashed, the internal watchdog mechanism restarts the editor and fires the [ready](#ready) event.

Reviewer: ```suggestion
Fired when the editor crashes (except of crashes during the editor initialization). Once the editor is crashed, the internal watchdog mechanism restarts the editor and fires the [ready](#ready) event.
```
